### PR TITLE
Fix #4193 to ban com.github.stephenc.jcip:jcip-annotations

### LINKED ISSUE #4193: Ban `com.github.stephenc.jcip:jcip-annotations`
Follow up on https://github.com/apache/camel-quarkus/pull/4183#issuecomment-1282400106.

### PR DESCRIPTION
**Pull request by zhfeng**

### CODE REVIEW

**Review by oscerd — APPROVED** [review]

**Review by jamesnetherton — APPROVED** [review]

**Review by aldettinger — APPROVED** [review]

File: pom.xml [hunk 1]
```diff
@@ -699,6 +699,7 @@
                                                 <exclude>ch.qos.logback:logback-access</exclude><!-- dismiss; the functionality is provided by JBoss Log Manager provided via quarkus-core -->
                                                 <exclude>ch.qos.logback:logback</exclude><!-- dismiss; the functionality is provided by JBoss Log Manager provided via quarkus-core -->
                                                 <exclude>com.github.fge:*</exclude><!-- Use com.github.java-json-tools:* instead -->
+                                                <exclude>com.github.stephenc.jcip:jcip-annotations</exclude>
```
Comment (ppalaga) [inline pom.xml:702]:
```suggestion
                                                <exclude>com.github.stephenc.jcip:jcip-annotations</exclude><!-- Should be needed for neither compilation nor at runtime -->
```

**Review by ppalaga — APPROVED** [review]
A comment suggestion inline, otherwise perfect. Thanks @zhfeng !

**Comment by ppalaga** [comment]
I am working on reusing banned dependencies XML files distributed by Quarkus. Merging this would conflict, so let me adopt this change in my branch.

**Comment by ppalaga** [comment]
Replaced by https://github.com/apache/camel-quarkus/pull/4441

**Pull request closed without merging**